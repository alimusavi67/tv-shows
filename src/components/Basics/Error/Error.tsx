import React from 'react';

interface ErrorProps {
  message: string;
}

const Error: React.FC<ErrorProps> = ({ message }) => {
  return (
    <div className="flex justify-center items-center h-full py-8">
      <div className="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative max-w-md mx-auto">
        <strong className="font-bold">Error!</strong>
        <span className="block sm:inline ml-2">{message}</span>
      </div>
    </div>
  );
};

export default Error;
