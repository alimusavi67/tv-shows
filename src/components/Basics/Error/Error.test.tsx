import React from 'react';
import { render } from '@testing-library/react';
import Error from './Error';

test('renders Error correctly', () => {
  const { asFragment } = render(<Error message="An error occurred!" />);
  expect(asFragment()).toMatchSnapshot();
});
