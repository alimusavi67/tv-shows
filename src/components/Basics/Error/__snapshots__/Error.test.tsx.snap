// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error correctly 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center h-full py-8"
  >
    <div
      class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative max-w-md mx-auto"
    >
      <strong
        class="font-bold"
      >
        Error!
      </strong>
      <span
        class="block sm:inline ml-2"
      >
        An error occurred!
      </span>
    </div>
  </div>
</DocumentFragment>
`;
